core/sql/lib_mgmt/src/main/java/org/trafodion/libmgmt/FileMgmt.java

```diff
@@ -547,6 +558,64 @@ private static void checkFile(String fname, int dataSize) throws SQLException {
 			throw new SQLException("Jar file size is over the threshold[100Mb]");
 		}
 	}
+	
+	public static void connectBy(String columnName, String originSql, String startWithSql, String startWithKey,ResultSet[] rs) throws SQLException{
+		String[] out =new String[1];
+		getStartWith(startWithSql, startWithKey, out); 
+		String[] s = out[0].replaceAll("^\\[|\\]$", "").split("\\s*,\\s*");
+		String clause = " ";
+		for(int i = 0;i<s.length;i++){
+			if(i == s.length -1){
+				clause += columnName + "=\'" + s[i] + "\'";
+			}else{
+				clause += columnName + "=\'" + s[i] + "\' or ";
+			}
+		}
+		String connectSql = String.format(originSql, clause);
+		Connection conn = getConn();
+		PreparedStatement ps = conn.prepareStatement(connectSql);
+		try{
+			rs[0] = ps.executeQuery();
+		} catch(SQLException e){
+			LOG.error(originSql,e);
+			throw e;
+		} 
+	}
+	
+	
+	public static void getStartWith(String sql, String startWithKey, String[] out) throws SQLException{
+		String sqlTest = String.format(sql, startWithKey);
+		int searchLevel = 0;
+		ArrayList<String> ids = new ArrayList<String>();
+		Connection conn = getConn();
+		PreparedStatement ps = conn.prepareStatement(sqlTest);
+		try{
+			ids.addAll(bfsSearch(sql,startWithKey,searchLevel,ps));
+			out[0]= Arrays.toString(ids.toArray());
+		} catch(SQLException e){
+			LOG.error(sql,e);
+			throw e;
+		} 
+	}
+	
+	private static ArrayList<String> bfsSearch(String sql, String startWithKey,int searchLevel, PreparedStatement ps) throws SQLException{
```

Comment:
@zellerh , the SPJ can now create volatile table, that bug is already fixed recently in T2 driver.

Agree with Han's opinion. using a table to hold intermediate result is more generic. And the current solution is still serial, or did I misunderstand? 

After getting all the start seeds, it should be able to spawn multiple threads to do it in parallel, but using volatile table and using SQL's parallel infrastructure seems a better approach.

It is not easy to write a generic one, but this will be a very useful SPJ, so thanks for the effort.